src/components/Menu/MenuItem.tsx

```diff
@@ -93,7 +93,7 @@ class MenuItem extends UIComponent<any, any> {
         {childrenExist(children) ? (
           children
         ) : (
-          <a
+          <button
```

Comment:
Tested with the change as button and now it works as expected for each case. 
Space and Enter trigger onclick method. 
Tested with:
Chrome > JAWS with and without virtual cursor
Edge > Narrator when scan mode is on/off
Firefox > NVDA > when browse mode is on/off
OSX > Chrome > VoiceOver > without and with VoiceOver keys

Reply:
Today we have been testing more using "a" tag for menu and figure out following:

1.
When added on menu "a" handleKeydown method:
handleKeydown = e => {
    if (e.which === 32 || e.which === 13) {
      this.handleClick(e)
      // _.invoke(this.props, 'onClick', e, this.props)
      e.preventDefault()
    }
  }

2. Then space/enter was working properly

3. And JAWS was not listed link in link view... What is fine... 

So in the end I think we could go with "a" tag in menu. 

@jurokapsiar @miroslavstastny